usr/share/rear/output/default/940_grub2_rescue.sh

```diff
@@ -173,27 +176,39 @@ if is_true $USING_UEFI_BOOTLOADER ; then
     ) > $grub_config_dir/rear.cfg
 
     # Create rear.efi at UEFI default boot directory location.
+    # The build_bootx86_efi errors out if it cannot make a bootable EFI image of GRUB2:
     build_bootx86_efi $boot_dir/efi/EFI/BOOT/rear.efi $grub_config_dir/rear.cfg "$boot_dir" "$UEFI_BOOTLOADER"
 
-    # If UEFI boot entry for "Relax-and-Recover" does not exist, create it.
-    # This will also add "Relax-and-Recover" to boot order because if UEFI entry is not listed in BootOrder,
+    # If UEFI boot entry for 'Relax-and-Recover' does not exist, create it.
+    # This will also add 'Relax-and-Recover' to boot order because if UEFI entry is not listed in BootOrder,
     # it is not visible in UEFI boot menu.
-    if [[ $(efibootmgr | grep -cw $grub_rear_menu_entry_name) -eq 0 ]]; then
-        # This part might not go that well with drivers like HPEs cciss...
+    if efibootmgr | grep -q $grub_rear_menu_entry_name ; then
+        LogPrint "Skip creating new 'Relax-and-Recover' UEFI boot entry (it is already there)"
+    else
+        # This part might not go that well with drivers like HPEs cciss ...
         # However UEFI booting is present since Gen8 (AFAIK), and cciss drivers were replaced by hpsa long time ago,
         # so it looks like impossible configuration, lets wait ...
-        efi_disk_part=$(grep -w /boot/efi /proc/mounts | awk '{print $1}')
-        efi_disk=$(echo $efi_disk_part | sed -e 's/[0-9]//g')
-        efi_part=$(echo $efi_disk_part | sed -e 's/[^0-9]//g')
-
```

Comment:
this part should be eventually converted to use helper functions in layout-functions.sh (like 670_run_efibootmgr.sh)

Reply:
also it probably won't work properly in case of ESP on MD RAID (time to introduce more shared helpers instead of copying code from PR #2608 I guess...)

Reply:
For now I added a TODO comment to have it at least described.
I cannot do a the needed complete cleanup here now.